cmake/Modules/CocosBuildHelpers.cmake

```diff
@@ -109,7 +109,7 @@ function(cocos_copy_target_dll cocos_target)
   foreach(single_target_dll ${all_depend_dlls} )
```

Comment:
I do not like function get_target_depends_ext_dlls.
I think that better would be to have and maintain list of dlls.

Reply:
I think `get_target_depends_ext_dlls` is the most complicated function in those cmake files of cocos2d-x, it's a little difficult to understand the logic of collect dlls.

maintain a list of dlls is simple and easy to realize, and then how many dlls list should we supply? 3 dlls list, one dlls list for per language? but it should have different dlls when a win32 app use different libs, for as we all know one dll for one lib. 

I catch the logic of one dll for one lib, the app target need a dll if it use the corresponding lib. the current collect dlls logic is:
 
1. pick corresponding dll for a specific lib in function `cocos_find_dlls_for_lib`, finilly save it in `${_prefix}_DLLS`
 1. add a target property `DEPEND_DLLS` record a target depend dlls, collect dlls in function `cocos_use_pkg`
1. `get_target_depends_ext_dlls` depend on the target property `LINK_LIBRARIES` find a target depend libs, then collect dlls through the the target property `DEPEND_DLLS`. you know lib1 link lib2, lib2 may link lib3, lib3 may link lib4...., it's a tree struct, so I have to traverse through recursion.

it's so complicated, but I think it's quite closer the truth of a app use dlls.

Reply:
But how many dll list do we have to maintain? Android and Linux do not use dll (only static libs). Do we use dlls for Mac/iOS?

Reply:
dlls only need on windows. should we copy all the dlls for a application? I don't think so. 

For example when we build a cpp application, we don't need copy `lua51.dll`. it's not necessary. then we should have different dll list when different language. so 3 dlls list at least. 

How about for `websockets.dll` if a app don't use the websocket lib? so I think we need copy a dll only we the app use the corresponding lib.

Reply:
OK. I do not insist.

Reply:
As I said I do not insist but if it is only for win32 I think that it is fine to copy everything :)
I think that would be nice to setup for 3rdparty libraries CMake and use static libraries everywhere (even for win32)

Reply:
How dll copying works right now (with Visual Studio solution) for win32?

Reply:
it works before, and I said this commit  https://github.com/cocos2d/cocos2d-x/pull/18722/commits/a31ef36f42c0ae3371d10a13bf4edea908a834ae in this PR fixed https://github.com/cocos2d/cocos2d-x/issues/18723